frontend/src/routes/manage/Realm/Content/Edit/EditMode/index.tsx

```diff
@@ -100,6 +101,14 @@ export const EditModeForm = <FormData extends object, ApiData extends object>(
     const { id } = blocks[index];
 
 
+    const router = useRouter();
+    useEffect(() => (
+        router.listenAtNav(() => {
+            onCancel?.();
+        })
+    ));
```

Comment:
Can't you just do this? 

```suggestion
    useEffect(() => () => onCancel?.());
```